oop1-construct.js

```diff
@@ -20,58 +66,99 @@ Create a class that can categorize at least 6 of the creatures that you recall o
 
 class Shape {
     //your code here...
+    constructor(name, sides, base, height, area, length, width) {
+        this.name = name;
+        this.sides = sides;
+        this.base = base;
+        this.height = height;
+        this.area = area;
+        this.length = length;
+        this.width = width;
+    }
+
+
 }
+class Triangle extends Shape {
+    constructor(name, sides, base, height, area) {
+        super(name, sides, base, height, area);
+    }
+
 
+    HeronsCalc() {
+        let heronsStart = this.sides.reduce((accumulator, currentValue) => accumulator + currentValue, 0);
+        let variableS = heronsStart / 2;
+        let triangleArea = Math.sqrt(variableS * ((variableS - this.sides[0]) * (variableS - this.sides[1]) * (variableS - this.sides[2])));
+        let triangleHeight = 2 * (triangleArea / this.sides[0]);
+        this.height = triangleHeight;
+        this.base = this.sides[0];
+        this.area = triangleArea;
```

Comment:
Refactor this to be modular and readable. Rememer it's convention to make our methods camelCase, so heronsCalc. Only Constructors and Classes are capitalized. Instead of heronsStart, what does that variable actually represent? The sum of the sides! So call it that! `sidesSum` or even `sumOfSides`. variableS is especially confusing because it looks like a typo of "variables". Call it `semiperimeter`. And once you calculate the next step, you have the area of the triangle. Since that's a field we want to save anyway, there's no sense in making an extra variable for it. Just set it equal to this.area right there, and use this.area in the next formula. Same with this.height on line 91.